packages/launcher-component/src/flows/launch-flow.tsx

```diff
@@ -20,21 +21,23 @@ export function useAutoSetDestRepository(defaultName: string, setApp) {
   const [showForm, setShowForm] = useState(false);
   const [loading, setLoading] = useState(true);
   useEffect(() => {
-    client.gitInfo().then(info => {
-      setApp((prev: ExampleApp | NewApp) => {
-        if (prev.destRepository.userRepositoryPickerValue && prev.destRepository.userRepositoryPickerValue.name) {
-          return prev;
-        }
-        return {...prev, destRepository: {userRepositoryPickerValue: {name: defaultName}, isProviderAuthorized: true}};
-      });
-      setShowForm(true);
-      setLoading(false);
-    }).catch(e => {
-      setApp((prev: ExampleApp | NewApp) => {
-        setShowForm(false);
-        return {...prev, destRepository: {isProviderAuthorized: false}};
-      });
-      setLoading(false);
+    gitInfoLoader(client).then(gitInfo => {
+      if (gitInfo.login) {
+        setApp((prev: ExampleApp | NewApp) => {
+          if (prev.destRepository.userRepositoryPickerValue && prev.destRepository.userRepositoryPickerValue.name) {
+            return prev;
+          }
+          return {...prev, destRepository: {userRepositoryPickerValue: {name: defaultName}, isProviderAuthorized: true}};
+        });
+        setShowForm(true);
+        setLoading(false);
+      } else {
+        setApp((prev: ExampleApp | NewApp) => {
+          setShowForm(false);
+          return {...prev, destRepository: {isProviderAuthorized: false}};
+        });
+        setLoading(false);
+      }
```

Comment:
@edewit I don't get why you didn't use the catch block (and checked for 404) for this one?